### ISSUE TITLE
COM Exception when OGCS starts on startup + no log file

### ISSUE THREAD

**Issue by dfiord** [phw198/OutlookGoogleCalendarSync#263]
Upon loading I received a COM exception error and was told to open an issue and upload the log file. I am doing so, It will not let me attache the log file as a .log file I am attaching it as a .txt file.
Thank you.

[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1129557/OGcalsync.txt)

**Comment by phw198** [comment]
Was the massage you got "COM exception encountered. Please log an Issue on GitHub and upload your OGcalsync.log file."? If so, I don't see any evidence of this in the log file you provided.

Does the error always happen / repeatable?

**Comment by m-rm** [comment]
I actually also have this issue.

![ogcs](https://user-images.githubusercontent.com/7124804/28056794-87aeabb0-661e-11e7-9e64-663af8b239a8.png)

No log is written when this happens. Thats why all similar issues reported here show nothing in their OGcalsync.txt. It is repeatable: For me it happens when "Start on login" is activated. If I deactivate this option and put a shortcut in the startup folder instead it works fine.

Edit: When choosing "Debug" after the application crashes Visual Studio provides this info: 
```
System.NullReferenceException was unhandled
Message: An unhandled exception of type 'System.NullReferenceException' occurred in OutlookGoogleCalendarSync.exe
Additional information: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
```

The German text means "The object reference was not set to an object instance.".

**Comment by phw198** [comment]
OK, I'm rather mystified why there's no logging of this - it should have all the start up information and then one of the last lines would be "Starting a new instance of Outlook."

However, delaying the startup has been requested elsewhere (#178) and also the cause of another COM error (#239) so I will try and implement this soon.

**Comment by m-rm** [comment]
I think I found the Problem. I didn't know that OGCS starts Outlook, so I had put Outlook in startup to start it automatically. This means Outlook opened first and then OGCS. For testing I removed Outlook from startup and now it starts without this issue.

**Comment by phw198** [comment]
OK, useful to know....though it shouldn't make any difference whether it's OGCS that starts Outlook, or whether it connects to an already running instance of Outlook.

Could you run a search of your entire hard-drive for any other `OGcalsync.log` files and see if it's been written to somewhere odd on machine startup (perhaps an environment variable isn't instantiated properly when it starts)?!

Also, is this new behaviour? Has the problem started in a recent release?

**Comment by m-rm** [comment]
Windows couldn't find any other OGcalsync.log.

I can't say if it's new behavior, I only started using OGCS a month ago.

**Comment by dfiord** [comment]
Hi Paul,

Thank you for following up with me on this issue.

There are 3 .log files in the directory: C:\Users\Don\AppData\Roaming\Outlook Google Calendar Sync . Two of them have dates following the file name and one does not. I am sending you the two most recent files.

Please let me know if there is anything else you might need.

Thanks again.
Don

**Comment by dfiord** [comment]
Hi Paul,

Interesting side note, I took the advice of m-rm regarding the process of OGCS starting Outlook and like him had an Outlook shortcut in the startup folder. I removed this shortcut and did a restart and did not get the error message.  Interestingly, OGCS did not start Outlook.  So, I put the shortcut for Outlook bck into the Startup folder and restarted and DID NOT get the error?  I have tried restarting a number of times since and the error has not reappeared.  

Following this I also upgraded to 2.5.1.0 and did restarts and have had no problem. 

Thanks for your help.  BTW I also opened #262 and was wondering if you had a chance to lok t that one.  That is extremely frustrating.  Thanks again and lease let me know if I can help.

**Comment by phw198** [comment]
Well it just keeps getting stranger! FWIW, when OGCS "starts" Outlook it doesn't start the application in full GUI mode, but you'll see an icon in the system tray saying an application is accessing Outlook. You may also get a reminders pop-up if you have any Outlook reminders due.

Still feels like putting a delay in the start of OGCS may help things.

**Comment by dfiord** [comment]
Well, I just restarted and I received the error again. I am confused.

I am attaching my log file for your review.

**Comment by phw198** [comment]
:package: Attached is a pre-release build [v2.5.1.2.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1150620/v2.5.1.2.zip) - apply in the same way as a hotfix and make sure you are running v2.5.1 before patching. [See instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix).

There is a new setting next to the "Start on login" checkbox. Please let me know how it goes.

**Comment by dfiord** [comment]
Hi Paul, 

Looks light you have been burning the Midnight Oil. 

The Hot Patch seems to work fine with a 20 second delay on my system.  Thanks.

**Comment by phw198** [comment]
In fact, I've found the bug that's causing log files to not generate too.

:package: [v2.5.1.3.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1150707/v2.5.1.3.zip) - please ensure you are running at least v2.5.1 before applying [[see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)].

I think that wraps this issue up. Thanks for quick testing @dfiord.

**Comment by dfiord** [comment]
Thanks again. BTW, to make a donation to your software development effort, it only allows you to enter, I guess it is Lira. That doesn’t help us here in the USA.

**Comment by phw198** [comment]
@dfiord Thanks for looking to make a donation! Not sure what's happening there - Lira's not much use to me either!!

Can you provide a screenshot of what you're seeing? How are you getting to the donation screen? Do you have a PayPal account or you're just needing to donate from a card?

**Comment by m-rm** [comment]
Thanks for the hotfix. 10 seconds of delay resolves this issue. Since you also fixed the log file not generating I can now upload a log of what happens with 0 seconds delay.

[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1152111/OGcalsync.txt)

Thanks for fixing this so fast!

**Comment by phw198** [comment]
Thanks @m-rm that's really useful. Looks like you were getting `RPC_E_CALL_REJECTED`, so I could back off and retry under that scenario too...

**Comment by dfiord** [comment]
When you select the Donate button from the About screen this is what you get. Note the type of currency next to Donation amount, I guess that is British Pounds.

This is probably working the way it should but, I would prefer to donate in USD and have the conversion made on the other side. Just my preference I guess. When I select the country I am donating from, i.e. United States, the currency does not change and I do not know how many British Pounds equals the amount I want to donate in USD. Example, $10.00 USD = how many GBP?

Thanks.
Don

**Comment by phw198** [comment]
@dfiord Were there meant to be screenshots in your reply, as they don't come through when updating GitHub via an email :disappointed: 

I'm not sure it makes too much difference where the conversion happens - either your bank/credit card will charge, or PayPal will. But if you have a PayPal account, then log in and go to the "Send Money" option (to donate@paulwoolcock.co.uk) - this way you can choose the currency. [This is a good exchange rate site](http://www.xe.com/currencyconverter/convert/?Amount=10&From=USD&To=GBP) - currently 10USD is 7.66GBP...

**Comment by phw198** [comment]
:package: @m-rm Would appreciate it if you could test the attached [v2.5.2.1.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1167480/v2.5.2.1.zip) with a delay of 0 seconds to see if that is now handled better.

Cheers!

**Comment by m-rm** [comment]
I tested it: It throws the same error as before. See the attached log file.

[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1169526/OGcalsync.txt)

**Comment by phw198** [comment]
:hankey: Stupid typo in the code, `0x80080001` instead of `0x80010001`.

:package: Could I trouble you once more for [v2.5.2.9.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1180747/v2.5.2.9.zip)?

**Comment by m-rm** [comment]
Looks good, restarted 3 times with 0 seconds delay without getting any error message.

**Comment by mpking828** [comment]
Just jumping on the Me too train.
For me, it happens every morning when I start my computer up..  But not when I reboot. (Took me a while to catch onto that pattern).  I think it's Outlook syncing with the server that is the difference.

I'm on V2.5.0.0
Here's the last log I got (I don't have the logging patch, so I expect nothing in there)

I'm gonna grab 2.5.2.9 an see how tomorrow goes.

**Comment by phw198** [comment]
I'm going to close this issue as pertaining to the missing logfile as solved; will track the various other COM errors on separate issues.

@mpking828 would be interested to know if v2.5.2.9 fixes your issue, if you leave a comment here even after closed. We're up to v2.5.2.11 on another ticket...

**Comment by mpking828** [comment]
I just loaded v2.5.2.9 this morning. I'll let you know if it goes sideways.

**Comment by dfiord** [comment]
Hi Paul,

Since I have been on this thread with the original issue, I installed 2.5.2.9 also.  Unfortunately, I have run into a problem with running this and the immediately previous version, although the previous version I only experienced this problem once.

When I run this version on startup, with a 20 second delay, I get an error message stating, Outlook and OGCS are running in different security elevations. Both must be running in Standard or Administrator mode.  Although this is disturbing and concerning, the interesting issue is the when I run OGCS again it starts up fine, with no error message.

I just wanted you to be aware of this.  Any advice would be appreciated.

Don
